src/gui/widgets/TrackLabelButton.cpp

```diff
@@ -107,8 +109,12 @@ void TrackLabelButton::renameFinished()
 	if( !( ConfigManager::inst()->value( "ui", "compacttrackbuttons" ).toInt() ) )
 	{
 		m_renameLineEdit->hide();
-		setText( m_renameLineEdit->text() );
-		m_trackView->getTrack()->setName( m_renameLineEdit->text() );
+		if( m_renameLineEdit->text() != text() )
+		{
+			Engine::getSong()->setModified();
+			setText( m_renameLineEdit->text() );
+			m_trackView->getTrack()->setName( m_renameLineEdit->text() );
```

Comment:
I think you should move the ```Engine::getSong()->setModified();``` to here so it's the last in the if(), like on line 90-91^